backend/src/apiserver/resource/resource_manager.go

```diff
@@ -384,8 +384,13 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 		return nil, util.NewInternalServerError(err, "Failed to create a workflow for (%s)", workflow.Name)
 	}
 
+	workflowFinalSpecManifestBytes, err := json.Marshal(workflow)
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to marshal final workflow spec manifest into JSON")
+	}
+
 	// Store run metadata into database
-	runDetail, err := r.ToModelRunDetail(apiRun, runId, util.NewWorkflow(newWorkflow), string(workflowSpecManifestBytes))
+	runDetail, err := r.ToModelRunDetail(apiRun, runId, util.NewWorkflow(newWorkflow), string(workflowFinalSpecManifestBytes))
```

Comment:
@chensun @IronPan Can you help me review if this is reasonable?
I want the patched service account to be queryable later in DB. Does it make sense if we store the patched version of workflow spec manifests into the DB instead? I cannot think of any problems, but maybe I can miss something.

(discussed offline with @jingzhang36, we think this is okay)

Reply:
(I'm adding this change intentionally without fixing unit tests, I'd like some early feedback first.)

Reply:
I can't think of any issue. In fact, I was also puzzled why didn't we save the final spec instead...

Reply:
Ideally, we should split this change, one for storing the final spec instead of the template one, the other for setting a different SA. But you feel free to decide whether it's worth the effort.

Reply:
I want to save some separate PRs... it's a little troublesome keeping track of a lot of PRs (especially PRs depending on PRs)